src/mon/OSDMonitor.cc

```diff
@@ -5797,6 +5797,21 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
+int OSDMonitor::check_pg_num(int pg_num, int size, ostream *ss)
+{
+  int64_t max_pgs_per_osd = g_conf->mon_pg_warn_max_per_osd;
+  int64_t max_pgs = max_pgs_per_osd * osdmap.get_num_osds();
```

Comment:
counting only up+in osds might be useful if a bunch of osds are not running but not removed from crush